x/mongo/driver/topology/server.go

```diff
@@ -663,7 +664,7 @@ func (s *Server) update() {
 			s.rttMonitor.connect()
 		}
 
-		if isStreamable(s) || connectionIsStreaming || transitionedFromNetworkError {
+		if isStreamable(s) && (serverSupportsStreaming || connectionIsStreaming) || transitionedFromNetworkError {
```

Comment:
The logic for this was not implemented correctly, see specifications here: https://github.com/mongodb/specifications/blame/master/source/server-discovery-and-monitoring/server-monitoring.rst#L747